packages/builder/builder/src/plugins/arco.ts

```diff
@@ -0,0 +1,45 @@
+import { isPackageInstalled } from '@modern-js/utils';
+import { useSSR } from './antd';
+import type { DefaultBuilderPlugin } from '@modern-js/builder-shared';
+
+export const builderPluginArco = (): DefaultBuilderPlugin => ({
+  name: `builder-plugin-arco`,
+
+  setup(api) {
+    const ARCO_NAME = '@arco-design/web-react';
+    const ARCO_ICON = `${ARCO_NAME}/icon`;
+
+    api.modifyBuilderConfig(builderConfig => {
+      const { transformImport = [] } = builderConfig.source || {};
+
+      if (
+        transformImport === false ||
+        !isPackageInstalled(ARCO_NAME, api.context.rootPath)
+      ) {
+        return;
+      }
+
+      const isUseSSR = useSSR(api.context.target);
+
+      if (!transformImport?.some(item => item.libraryName === ARCO_NAME)) {
```

Comment:
The official Arco plugin @arco-plugins will injects `transformImport` according to its own options. The @arco-plugins must execute after the plugins of @modern-js, which may result in multiple injections of `transformImport`. It is unlikely that the injection behavior of @arco-plugins can be changed. Should we consider not handling it here?

Reply:
The official plugin may not handle SWC / Rspack scenarios, which can be handled by `source.transformImport`.

Reply:
And the above logic can not be removed from Modern.js / EdenX now,  because it existed in the previous version of EdenX.

Reply:
@JSerFeng cc

Reply:
Yes, I think the duplicated configurations will not affect the build result, maybe will slow down the build speed a bit, which is acceptable.

Anyway, if it will cause any issues,  we can add some logics to handle it in the future.